doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,892 @@
+    }
+},
+```
+
+```
+; Existing table ACL_RULE_TABLE
+; ACL action for disabling ARS
+
+
+key                     = ACL_RULE|ACL_TABLE_NAME|RULE_NAME
+
+;field                  = value
+
+DISABLE_ARS_FORWARDING  = boolean                       ;ARS operation disabled on matching packets
+
+Configuration example:
+
+"ACL_RULE": {
+    "MY_ACL_1|NO_ARS" : {
+        "SRC_IP": "10.2.130.0/24",
+        "DISABLE_ARS_FORWARDING" : "true"
+    }
+}
+```
+
+
+### Counters
+
+Following counters defined in SAI and will be supported via FlexCounters in phase 2:
+
+| Level            | Supported SAI counters  |
+| ---------------- | ----------------------- |
+| lag              | SAI_LAG_ATTR_ARS_PACKET_DROPS<br>SAI_LAG_ATTR_ARS_PORT_REASSIGNMENTS
+| nexthop group    | SAI_NEXT_HOP_GROUP_ATTR_ARS_PACKET_DROPS<br>SAI_NEXT_HOP_GROUP_ATTR_ARS_NEXT_HOP_REASSIGNMENTS<br>SAI_NEXT_HOP_GROUP_ATTR_ARS_PORT_REASSIGNMENTS|
+
+Flex counter group will be created for each level.
+
+### CLI changes
+
+1. ARS LAG counter polling:
+```
+    counterpoll ars_lag <enable | disable >
+```
+
+2. ARS LAG counter polling interval
+```
+    counterpoll ars_lag inteval < interval-val-ms >
+```
+
+3. ARS LAG counters show
+```
+    show ars_lag_counters [name]
+```
+
+4. ARS NHG counter polling:
+```
+    counterpoll ars_nexthop_group <enable | disable >
+```
+
+5. ARS NHG counter polling interval
+```
+    counterpoll ars_nexthop_group inteval < interval-val-ms >
+```
+
+6. ARS NHG counters show
+```
+    show ars_nexthop_group_counters [vrf <vrf-name>] [prefix <prefix>]
+```
+
+* Show examples
+    * show ars_lag_counters PortChannel001
+```
+ Name             Drops    Port reassignments
+---------------   -----    ------------------
+ PortChannel001     10           100
+```
+
+    * show ars_nexthop_group_counters default 192.168.0.10/24
+```
+ Name                          Drops   Nexhop reassignments     Port reassignments
+--------------------------     -----   ---------------------    -------------------
+ default | 192.168.0.10/24      10             50                      100
+```
+
+### Warmboot and Fastboot Design Impact
+During warmboot or fastboot, both ARS and ACL rules configurations are restored from the CONFIG_DB.
+Counter polling is delayed at system startup.
+
+
+#### Restrictions/Limitations
+Implementation will be done in two phases. 
+1. Phase 1 will support: 
+    - Primary path only
+    - Quality parameters
+2. Phase 2 will support:
+    - Alternative path
+    - LAG support
+    - Statistics
+
+
+#### Unit Test cases  
+
+Tests separated into two groups - mandatory and optional (only if supported by vendor sai) parts.
+
+- Mandatory:
+
+1. ARS Profile Creation
+    * Verify that ARS profiles are created successfully with valid parameters.
+    * Test error handling for invalid profile configurations.
+
+2. Enabling ARS on Ports
+    * Ensure ARS is enabled on specified ports.
+    * Validate behavior when enabling ARS on unsupported interfaces.

```

Comment:
via enabling ARS on NHG when it is defined on not-supported interfaces ? Missing logic in the design